### ISSUE TITLE
date picker fails when wordpress language isn't set to English.

### ISSUE THREAD

**Issue by yivi** [WebDevStudios/CMB2#267]
The configuration for my date field is this:

```
$main->add_field(
		[
			'name' => __( 'Fecha', 'ojiva-plugin' ),
			'id'   => $prefix . 'fecha_realizacion',
			'type' => 'text_date_timestamp',
			'date_format' => "d/m/Y",
		]
	);
```

If WP is set to English...
![general_settings_ _ojiva_ _wordpress](https://cloud.githubusercontent.com/assets/1815039/6870482/a0c7c214-d49c-11e4-9308-d0d72c8d96a6.png)

...then the field works, although it ignores my `date_format` settings (should be 04/03/2015):
![editar_portfolio_ _ojiva_ _wordpress](https://cloud.githubusercontent.com/assets/1815039/6870532/04bbdd8c-d49d-11e4-95c5-a37705cfbf16.png)

But at least, it works, displays incorrectly but saves correctly.

If I then change WP language to Spanish:
![ajustes_generales_ _ojiva_ _wordpress](https://cloud.githubusercontent.com/assets/1815039/6870558/2edd26ca-d49d-11e4-9aa0-896a8dde5836.png)

The date picker just wont work:
![editar_portfolio_ _ojiva_ _wordpress](https://cloud.githubusercontent.com/assets/1815039/6870581/4d81522c-d49d-11e4-867f-154486cff913.png)

It wont display the date correctly, nor save it in the DB.

**Comment by yivi** [comment]
Am I the only one having trouble with date_format options and with date pickers in diferent locales? :(

**Comment by jtsternberg** [comment]
@yivi You might try first taking this to [the forums](https://wordpress.org/support/plugin/cmb2).

**Comment by yivi** [comment]
Sorry, wp forums are horrible and hard to search/use, thought this was the proper place to report bugs and contribute back... Isn't it?

Anyway, I found what the problem is. There is a mistranslation in `cm2-es_ES.po`, line 362.

`mm/dd/yy` to `dd / mm / aa`, which doesn't make a lot of sense (`aa` isn't a valid `formatDate` [format] (https://api.jqueryui.com/datepicker/#utility-formatDate)), that probably happened because in CMB2_hookup.php it does:
`'dateFormat'      => __( 'mm/dd/yy', 'cmb2' ),`
Where it could probably do better by doing 
`'dateFormat'      => _x( 'mm/dd/yy', 'Valid formatDate string for jquery-ui datepicker','cmb2' ),`

I also saw there were a couple Spanish translations missing. I'll just make a PR with these fixes and additions.
Regards.
